### ISSUE TITLE
pew cannot be used as user

### ISSUE THREAD

**Issue by Vayel** [berdario/pew#124]
Installing virtualenv as user (`pip install --user virtualenv`) does not create the `virtualenv` command, so pew does not work. Maybe better to call `python virtualenv.py args` instead of `virtualenv args`?

Another solution (French): http://scylardor.fr/2013/06/09/zsh-command-not-found-virtualenv/

**Comment by Zopieux** [comment]
I'm not sure what's the issue there. If you install a package as user, it's stored in `~/.local`. As such, it is *normal* to put `~/.local/bin` in your `PATH` – that's the standard procedure. Doing so will make the `virtualenv` command available.

**Comment by Vayel** [comment]
You're right, but maybe there is a simple solution to make it work automatically for noobs?

**Comment by Zopieux** [comment]
Some installers might append the `export PATH="…:$PATH"` stuff into `.bashrc`, but if you ask me, I hate software that messes with my dotfiles without notice.

**Comment by bochecha** [comment]
> You're right, but maybe there is a simple solution to make it work automatically for noobs?

Most modern Linux distributions add `~/.local/bin` to the user `$PATH` by default.

Please ask your distribution to do the same.

**Comment by apatrushev** [comment]
Looks like that best solution for `virtualenv` problems is to call it as module:
```
python -m virtualenv ...
```
or in python code replace:
```python
check_call(["virtualenv", ...])
```
with:
```python
check_call([sys.executable, '-m', "virtualenv", ...])
```

**Comment by erinxocon** [comment]
I don't think we will mess with your path by default.  Closing as will not fix

**Comment by apatrushev** [comment]
Ok. As you wish. I just want to note that the problem with `virtualenv` is much wider than specified by current issue. Have you any **justified** arguments against my solution? I am just interested in. It is quite simple and will solve a lot of cases. I can provide you a bunch of examples where pew is currently fails, though can work with very simple fix described above.

**Comment by erinxocon** [comment]
@apatrushev Apologies, but I don't think our program should make changes to the PATH.  If you want to implement a fix you can submit a PR!  I'll let @berdario weigh in on this, but for now will leave it closed :)

**Comment by apatrushev** [comment]
@erinxocon Did you read my comment at least? I did not propose to make any changes in the PATH.

**Comment by erinxocon** [comment]
@apatrushev I did not, apologies was looking at this on mobile, I'll look into this.  If you know how to fix it feel free to submit a PR!

**Comment by berdario** [comment]
The `check_call([sys.executable, '-m', "virtualenv", ...])` fix seems to be quite simple

not sure why it has never been done before. Maybe this only works when virtualenv's location is in `sys.path`, and usually `sys.path` inherits from `PATH`. Thus I assumed that in almost no cases it would be possible to `import virtualenv` without having the `virtualenv` executable available

(I'm in a bit of a rush, so I haven't yet tried to do a `pip install --user virtualenv` )

Thanks!

**Comment by bochecha** [comment]
That might actually also fix #102. 😄

**Comment by apatrushev** [comment]
@bochecha That was exactly what I said above - in fact there are lots of cases where "module call" of `virtualenv` is much more stable and predictable than simple executable call.
